### PR TITLE
Starring

### PR DESCRIPTION
**Pull request by Meha-Agrawal**
I think it works now, pls test bc I might be going crazy...

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -179,9 +179,25 @@ export default class App extends Component<Props> {
       });
   };
 
-  updateAuthUser(newUser) {
-    this.setState({ auth: newUser });
-  }
+  toggleStarred = async (id, starDict) => {
+    let curItems;
+    if (this.state.starredItems == null) {
```
Comment (joel99) [inline App.js]: when would this ever be the case?

File: components/ButtonControl.js [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
+    this.updateIndex = this.updateIndex.bind(this);
+  }
 
-    updateIndex(selectedIndex) {
-        this.setState({ selectedIndex })
-        this.props.onChangeIndex(selectedIndex)
-    }
+  updateIndex(selectedIndex) {
+    this.setState({ selectedIndex });
+    this.props.onChangeIndex(selectedIndex, this.props.starredItems);
```
Comment (joel99) [inline components/ButtonControl.js]: do you think the selectedIndex should just be passed down along with a callback to change index? Doesn't seem to be a reason to hold state heres

File: components/ScheduleCard.js [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
+  StyleSheet,
+  TouchableHighlight,
+  Image,
+  TouchableOpacity
+} from "react-native";
 import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
 import { faPlusCircle, faStar } from "@fortawesome/free-solid-svg-icons";
 import { thisExpression } from "@babel/types";
 import TagList from "./TagList";
+import AsyncStorage from "@react-native-community/async-storage";
```
Comment (joel99) [inline components/ScheduleCard.js]: bad import?

File: screens/Schedule.js [hunk 4]
```diff
@@ -72,49 +80,54 @@ export default class Schedule extends Component<Props> {
...
           )
         );
+        this.starDict[curTalk.id] = false;
       }
     }
-
     return eventProps;
   };
 
+  starDict = {};
```
Comment (joel99) [inline screens/Schedule.js]: throw these random floating things into the constructor please

File: screens/Schedule.js [hunk 5]
```diff
@@ -13,6 +13,8 @@ import { SearchBar } from "react-native-elements";
 import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
 import Modal from "react-native-modal";
 import Event from "../components/events/Event";
+import AsyncStorage from "@react-native-community/async-storage";
```
Comment (joel99) [inline screens/Schedule.js]: i don't think async storage is necessary anywhere outside of `App`

File: components/ButtonControl.js (hunk 2 above)
Comment (Meha-Agrawal) [reply components/ButtonControl.js]: tru tru

File: App.js (hunk 1 above)
Comment (Meha-Agrawal) [reply App.js]: when asyncstorage first gets initialized (aka when somebody opens the app for the first time)

**Pull request merged**